generator/lib/src/generator.dart

```diff
@@ -1130,9 +1133,10 @@ You should create a new class to encapsulate the response.
             )
             .property('copyWith')
             .call([], {
-              _baseUrlVar: baseUrl.ifNullThen(
+              _baseUrlVar: refer('_combineBaseUrls').call([
                 refer(_dioVar).property('options').property('baseUrl'),
-              )
+                baseUrl,
+              ])
             })
```

Comment:
Upon further review of the code, it doesn't make much sense to keep this copyWith method. When an "absolute" path is passed to dio, it automatically ignores the baseUrl set in its BaseOptions. Therefore, it would be more appropriate to concatenate the baseUrl with the request path instead.